packages/next/build/webpack-config.ts

```diff
@@ -351,6 +351,11 @@ export default async function getBaseWebpackConfig(
     },
   }
 
+  if (config.experimental.modern) {
+    terserOptions.module = true
+    terserOptions.ecma = terserOptions.compress.ecma = terserOptions.output.ecma = 8
```

Comment:
This needs to be adjusted in the esm plugin for nomodule bundles (to use the old default).